tools/project_manager/external_catalog_search.php

```diff
@@ -129,186 +217,227 @@ function do_search_and_show_hits()
     $hits_per_page = 20; // Perhaps later this can be a PM preference or an option on the form.
-    $i = 1;
-    // PHPStan up to at least 1.10.57 has a bug where it doesn't correctly detect
-    // that $i is updated in the loop. We need to sprinkle several ignores to get
-    // it to ignore this.
-    // @phpstan-ignore-next-line
-    while (($start <= yaz_hits($id) && $i <= $hits_per_page)) {
-        $rec = yaz_record($id, $start, "array");
 
+    $marc_records = [];
+    for ($i = 0; $i < $hits_per_page && $start + $i <= $total_hits; $i++) {
+        $rec = yaz_record($id, $start + $i, "array");
         // if $rec isn't an array, then yaz_record() failed and we should
         // skip this record
         if (!is_array($rec)) { /** @phpstan-ignore-line */
-            $start++;
             continue;
         }
-
-        //if it's not a book don't display it.  we might want to uncomment in the future if there are too many records being returned - if (substr(yaz_record($id, $start, "raw"), 6, 1) != "a") { $start++; continue; }
         /** @phpstan-ignore-next-line */
         $marc_record = new MARCRecord();
         $marc_record->load_yaz_array($rec);
-
-        if ($i % 2 == 1) {
-            echo "<tr>";
+        if (!in_array($marc_record->get_type_of_record(), ["Language material", "Manuscript language material"])) {
+            $num_nontext++;
+            if ($hide_nontext) {
+                continue;
+            }
         }
+        $marc_records[] = [$rec, $marc_record];
+    }
+    yaz_close($id);
 
-        echo "<td class='center-align top-align' style='width: 5%;'>";
-        echo "<input type='radio' name='rec' value='".base64_encode(serialize($rec))."'>";
-        echo "</td>";
-        echo "<td class='left-align top-align' style='width: 45%;'>";
-        echo "<table class='basic' style='width: 100%;'>";
-
-        foreach ([
-            ['label' => _("Title"),     'value' => $marc_record->title],
-            ['label' => _("Author"),    'value' => $marc_record->author],
-            ['label' => _("Publisher"), 'value' => $marc_record->publisher],
-            ['label' => _("Language"),  'value' => $marc_record->language],
-            ['label' => _("LCCN"),      'value' => $marc_record->lccn],
-            ['label' => _("ISBN"),      'value' => $marc_record->isbn],
-        ] as $couple
-        ) {
-            $label = $couple['label'];
-            $value = $couple['value'];
-            echo "<tr>";
-            echo   "<th class='left-align top-align' style='width: 20%;'>$label:</th>";
-            echo   "<td class='left-align top-align'>$value</td>";
-            echo "</tr>\n";
+    // Display results
+    echo "<p>", sprintf(_("%d results returned."), $total_hits), "</p>\n";
+
+    if ($total_hits > 0) {
+        $encoded_fullquery = base64_encode(serialize($fullquery));
+        $url_base = "external_catalog_search.php?action=do_search_and_show_hits&fq=$encoded_fullquery&" . search_query_params();
```

Comment:
`$url_base` is output in the HTML directly so we either need to `&` -> `&amp;` here, or in the output function -- the latter is probably preferred. Ditto below where we continue constructing it.

Reply:
ACK. Sprinkled `attr_safe` where `$url_base` is used. Also I'd left an accidental (and now unnecessary) `if ($hide_nontext) { $url_base .= "&hide_nontext=on"; }` in the code.